cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/MaintenanceEventTopicReader.java

```diff
@@ -333,11 +331,7 @@ protected void ensureTopicCreated(Map<String, ?> config) {
    * @param maintenanceEventTopic Maintenance event topic.
    */
   protected void maybeCreateOrUpdateTopic(AdminClient adminClient, NewTopic maintenanceEventTopic) {
-    if (!createTopic(adminClient, maintenanceEventTopic)) {
-      // Update topic config and partition count to ensure desired properties.
-      maybeUpdateTopicConfig(adminClient, maintenanceEventTopic);
-      maybeIncreasePartitionCount(adminClient, maintenanceEventTopic);
-    }
+    KafkaCruiseControlUtils.maybeCreateOrUpdateTopic(adminClient, maintenanceEventTopic);
```

Comment:
same here, an unrelated (though maybe tempting) refactoring…

Reply:
This code is needed by the new code, but isn't available when written as a protected method. The protected method looks like it can be removed because it is only used in one place, but I left it since removing it could break any subclasses that override it. I could mark it as `@deprecated` instead.